src/app/services/xml-parsers/msdesc-parser.ts

```diff
@@ -0,0 +1,1594 @@
+        const { scope, extent, unit, quantity, atLeast, atMost, min, max } = attributes;
+
+        return {
+            type: Height,
+            content: parseChildren(xml, this.genericParse),
+            attributes,
+            scope,
+            extent,
+            unit,
+            quantity: quantity ? parseInt(quantity, 10) : undefined,
+            atLeast: atLeast ? parseInt(atLeast, 10) : undefined,
+            atMost: atMost ? parseInt(atMost, 10) : undefined,
+            min: min ? parseInt(min, 10) : undefined,
+            max: max ? parseInt(max, 10) : undefined,
+            gEl,
+        };
+    }
+}
+
+export class DimensionsParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+    private heightParser = createParser(HeightParser, this.genericParse);
+    private widthParser = createParser(WidthParser, this.genericParse);
+    private depthParser = createParser(DepthParser, this.genericParse);
+    private dimParser = createParser(DimParser, this.genericParse);
+
+    parse(xml: XMLElement): Dimensions {
+        const attributes = this.attributeParser.parse(xml);
+        const { dimensionsType, scope, extent, unit, quantity, atLeast, atMost, min, max } = attributes;
+        const heightEl = xml.querySelector<XMLElement>('scope > height');
+        const widthEl = xml.querySelector<XMLElement>('scope > width');
+        const depthEl = xml.querySelector<XMLElement>('scope > depth');
+        const dimEl = xml.querySelector<XMLElement>('scope > dim');
+
+        return {
+            type: Dimensions,
+            content: parseChildren(xml, this.genericParse),
+            attributes,
+            dimensionsType,
+            scope,
+            extent,
+            unit,
+            quantity: quantity ? parseInt(quantity, 10) : undefined,
+            atLeast: atLeast ? parseInt(atLeast, 10) : undefined,
+            atMost: atMost ? parseInt(atMost, 10) : undefined,
+            min: min ? parseInt(min, 10) : undefined,
+            max: max ? parseInt(max, 10) : undefined,
+            height: heightEl ? this.heightParser.parse(heightEl) : undefined,
+            width: widthEl ? this.widthParser.parse(widthEl) : undefined,
+            depth: depthEl ? this.depthParser.parse(depthEl) : undefined,
+            dim: dimEl ? this.dimParser.parse(dimEl) : undefined,
+        };
+    }
+}
+
+export class AcquisitionParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+
+    parse(xml: XMLElement): Acquisition {
+        const attributes = this.attributeParser.parse(xml);
+        const { notBefore, notAfter } = attributes;
+        // TODO: Add specific parser when name is handled
+        const name = Array.from(xml.querySelectorAll<XMLElement>(':scope > name'))
+        .map(e => parseChildren(e, this.genericParse));
```

Comment:
Please auto format the entire file: all new line starting with `.map` are not properly aligned